packages/block-editor/src/components/link-control/index.js

```diff
@@ -133,15 +139,18 @@ function LinkControl( {
 		withCreateSuggestion = true;
 	}
 
+	const setSettingsOpenWithPreference = useSelect( ( select ) => {
+		return select( blockEditorStore ).getSettings()
+			.setLinkControlAdvancedSettingsPreference;
```

Comment:
I updated so the setter is provided by the block editor settings. This means any editor can provide their own preference store rather than having these hard coded to WP only as was previously the case.